packages/editor/src/components/media-categories/index.js

```diff
@@ -0,0 +1,150 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { resolveSelect } from '@wordpress/data';
+import { decodeEntities } from '@wordpress/html-entities';
+
+/**
+ * Internal dependencies
+ */
+import { store as coreStore } from '@wordpress/core-data';
+
+const getExternalLink = ( url, text ) =>
+	`<a href="${ url }" target="_blank" rel="noreferrer noopener">${ text }</a>`;
+
+const getOpenvereLicense = (
+	license,
+	licenseVersion = '', // unknown version.
+	licenseUrl
+) => {
+	let licenseName = license.trim();
+	// PDM has no abbreviation
+	if ( license !== 'pdm' ) {
+		licenseName = license.toUpperCase().replace( 'SAMPLING', 'Sampling' );
+	}
+	// If version is known, append version to the name.
+	if ( licenseVersion ) {
+		licenseName += ` ${ licenseVersion }`;
+	}
+	// For licenses other than public-domain marks, prepend 'CC' to the name.
+	const isPublicDomainMark = [ 'pdm', 'cc0' ].includes( license );
+	if ( ! isPublicDomainMark ) {
+		licenseName = `CC ${ licenseName }`;
+	}
+	if ( !! licenseUrl ) {
+		license = getExternalLink(
+			`${ licenseUrl }?ref=openverse`,
+			licenseName
+		);
+	}
+	// TODO: add translators comment.
+	const markedLicence = isPublicDomainMark
+		? __( 'is marked with' )
+		: __( 'is licensed under' );
+	return `${ markedLicence } ${ license }`;
+};
+
+const getOpenverseCaption = ( item ) => {
+	const {
+		title = __( 'This work' ), // TODO: add translators comment.
+		foreign_landing_url: foreignLandingUrl,
+		creator,
+		creator_url: creatorUrl,
+		license,
+		license_version: licenseVersion,
+		license_url: licenseUrl,
+	} = item;
+	let _title = decodeEntities( title );
+	if ( !! foreignLandingUrl ) {
+		_title = getExternalLink( foreignLandingUrl, _title );
+	}
+	const fullLicense = getOpenvereLicense(
+		license,
+		licenseVersion,
+		licenseUrl
+	);
+	let _creator = decodeEntities( creator );
+	if ( creatorUrl ) {
+		_creator = getExternalLink( creatorUrl, creator );
+	}
+	_creator = ` ${ __( 'by' ) } ${ _creator }`; // TODO: add translators comment.
```

Comment:
Can `creator` be empty at any case? With some testing it seems the API returns `Unknown`.

Reply:
We shouldn't concat translated strings, we should use `sprintf` instead

Reply:
Yeah, I had left some translation things to fix, after having some Openverse folks validate the `caption` implementation. I guess I can do it now :)

Reply:
Hm.. so there are cases that the API doesn't even return `Unknown`, but an empty string or `undefined`? If that's the case I guess I should fallback to `Unknown`, right?

Reply:
Sorry for late reply, @ntsekouras. In Openverse frontend, we usually remove the "by ..." altogether, leaving just the media title and the license. Here's one example from the Science museum: https://search-staging.openverse.engineering/image/d9a41c10-3fce-429f-8993-5f44f0aaeea0 . 

_Note_: Actually, we might want to change the attribution on the Openverse end to include the creator of the picture instead of the creator of the object in the catalog. I [asked a question](https://wordpress.slack.com/archives/C02012JB00N/p1672890925520949) about it in the Openverse Make WordPress Slack. However, this shouldn't affect the implementation here.